lib/spack/spack/spec.py

```diff
@@ -129,7 +129,7 @@
     r"|"  # or
     # OPTION 2: an actual format string
     r"{"  # non-escaped open brace {
-    r"([%@/]|arch=)?"  # optional sigil (to print sigil in color)
+    r"([%@/]|[\w ][\w -]*=)?"  # optional sigil (or identifier or space) to print sigil in color
```

Comment:
Do we really mean to allow spaces in the middle of an attribute name? I would've expected 
```suggestion
    r"([%@/]|[\w ][\w-]*=)?"  # optional sigil (or identifier or space) to print sigil in color
```

Reply:
Personally I think it's fine to be more lenient here... you have the `=` which divides the key from the value -- do you really care what comes before?  I don't currently see a use for the extra space but I didn't think it was worth restricting what you label the value as. IMO it's up to whoever's formatting the string.